src/transformers/models/bigscience176b/modeling_bigscience176b.py

```diff
@@ -219,9 +239,17 @@ def forward(
         # ===========================
 
         # attention scores and attention mask [b, np, sq, sk]
-        attention_probs = self.scale_mask_softmax(attention_scores,
-                                                  attention_mask)
 
+        # softmax across tp ranks
+
+        aggregated_tensors = []
+        slices = self.num_heads / self.pretraining_tp
+        for i in range(self.pretraining_tp):
```

Comment:
add a quick comment or link to the pytorch repo issue on why we do this loop

Reply:
Fixed [here](https://github.com/younesbelkada/transformers/commit/a01a894a3d92779a000dcba0ca263de888df3c46)